tools/openblas_support.py

```diff
@@ -8,6 +8,7 @@
 import tarfile
 import textwrap
 import zipfile
+from packaging.tags import sys_tags
```

Comment:
It might be worthwhile adding an explicit requirement on `packaging` to [`build_requirements.txt`](https://github.com/numpy/numpy/blob/main/build_requirements.txt) and [`environment.yml`](https://github.com/numpy/numpy/blob/main/environment.yml)

Reply:
although I see this is run _after_ the `cibw_before_build.sh` script, so maybe a moot point.

Reply:
I removed the import and made this mod:

```
def get_linux(arch):
    # best way of figuring out whether manylinux or musllinux is to look
    # at the packaging tags. If packaging isn't installed (it's not by default)
    # fallback to assuming it's manylinux
    try:
        from packaging.tags import sys_tags

        tags = list(sys_tags())
        if 'manylinux' in tags[0].platform:
            return get_manylinux(arch)
        elif 'musllinux' in tags[0].platform:
            return get_musllinux(arch)
    except ImportError:
        return get_manylinux(arch)
```

Reply:
LGTM to do the try-except. A dependency on `packaging` that we don't actually have for `pip install .` is not desirable imho.

Reply:
I think you can use either `sys.implementation._multiarch` or sysconfig.get_config_var('MULTIARCH') without invoking packaging.

Reply:
On x86_64 this returns `'x86_64-linux-gnu'`

Reply:
On x86_64 alpine this returns `'x86_64-linux-gnu'` :-(

Reply:
These are the hoops that the `packaging` package has to go through to figure out if `musllinux` is being used: https://github.com/pypa/packaging/blob/main/src/packaging/_musllinux.py

Reply:
ahh, there is sysconfig.get_config_var('BUILD_GNU_TYPE'), which gives either `x86_64-pc-linux-gnu` or `x86_64-pc-linux-musl`

Reply:
packaging [uses this](https://github.com/pypa/packaging/blob/1b9da4420e9e23046dc095a3779f2607ae6e4d62/src/packaging/_musllinux.py#L32) to determine if it is running on musl linux. That feels so wrong.

Reply:
Using `packaging` might be more robust than checking `sysconfig.get_config_var('BUILD_GNU_TYPE')`?

Reply:
Feels like the `platform` or `sys` should have this information. It's also possible to get the tags from pip:
```
from pip._internal.models.target_python import TargetPython
target_python = TargetPython()
pep425tags = target_python.get_tags()
```